Model/Data/OrderUpdate.php

```diff
@@ -196,4 +196,31 @@ private function getOrderUrl(OrderInterface $order): string
     {
         return $this->storeManager->getStore()->getBaseUrl() . 'sales/order/view/order_id/' . $order->getEntityId();
     }
+
+    /**
+     * Validates the order data.
+     *
+     * @param Order $order
+     * @return bool|string True if validation passes, otherwise returns error message.
+     */
+    private function dataValidation(Order $order, $amwalOrderData)
+    {
+        // Define an associative array mapping Amwal order fields to their corresponding getter methods.
+        // This array is used for comparing Order object values with Amwal order data.
+        $fields = ['AmwalOrderId' => 'getId', 'RefId' => 'getRefId', 'BaseGrandTotal' => 'getAmount'];
+        try {
+            foreach ($fields as $orderMethod => $amwalMethod) {
+                $orderValue = call_user_func([$order, 'get' . $orderMethod]);
```

Comment:
I think both order and amwal order data are data objects, so we should be able to doo something like
```suggestion
                $orderValue = $order->getData($orderField);
```

Where $orderField would be the snake_case version of the fieldd